rfc/2022/2022-07-20 - Update Build Directory Structure of vets-website.md

```diff
@@ -0,0 +1,61 @@
+
+In the future, the subdirectories of isolated application bundles can be moved to `build/dist` to follow a more standard naming convention. 
+
+### Updating the content build for the new directory structure
+
+The `content-build` repository is responsible for creating the HTML files that are served on the site. By default, the build process in the `content-build` repository uses the `build/generated` path when injecting script tags that reference bundles and assets from `vets-website` deployments. This will need to be modified, since assets could be loaded from `build/generated` or one of the subdirectories for isolated application builds, e.g. `build/generated/vaos`. To better determine which paths in the deployment buckets to load bundles and assets from on each page, a generic script that handles this logic can be created. The build process in the `content-build` repository can then be updated to inject that generic script on each page. This essentially transfers control of determining which assets get loaded to `vets-website`, making it unlikely that production would break for applications migrating to the isolated structure.
+
+
+### Updating CI for the new directory structure
+
+Since there will no longer be a true *full build* once isolated application bundles are stored in their own subdirectory, we'll need to modify the process of building and testing global changes in CI. It won't be possible to build and test every application in one build anymore. One approach that would allow us to build and test multiple builds simultaneously involves Yarn Workspaces. With Yarn workspaces, each isolated application will have its own `package.json` file in the root of its directory, e.g. `src/applications/vaos/package.json`. The `package.json` files can be used to run the commands for each directory in `src/applications` that contains a `package.json`. This means we can use a command like `yarn workspaces foreach` at the root level to run commands like `build`, `test:unit`, `cy:run` (run Cypress tests), etc. The `workspaces foreach` command also has a `--parallel` option that can run the commands in parallel.
+
+## Risks
+
+One risk of this approach is that site wide elements rendered on pages of isolated applications can potentially get out of sync with site wide elements on non-isolated application pages. One example is if the header for the site has a major update that gets released in the daily production deploy. Since isolated applications would be excluded from the daily production deployment, the header update would not be applied to them at the time of the daily deployment, meaning some applications on the site would be using the new header, while others would be using the old header. This isn't ideal for consistency of the site, as we want all pages on the site to use the same site wide elements.
```

Comment:
Could this be addressed by relying more on the design system? If we had a `<va-header>` that was used in the template in the content build, I think that could eliminate the need for the `platform/site-wide` code that gets included in the app bundle.

Reply:
That's a good point. Having more site wide code in the design system could help remove that from the app bundles. We could still potentially run into issues with inconsistencies across apps though. If the version of the `component-library` was bumped in the root `package.json` and merged into `main`, isolated app commits merged into `main` afterwards would be deployed to prod using the new version of the `component-library`, whereas other non-isolated apps would still be using the old version of the `component-library` until the next daily production deploy happens